lua/entities/gmod_wire_expression2/core/custom/nav.lua

```diff
@@ -0,0 +1,348 @@
+local MAP_HAS_NAV = navmesh.IsLoaded()
+hook.Add("PlayerInitialSpawn","__e2_MAP_HAS_NAV",function()
+	MAP_HAS_NAV = navmesh.IsLoaded()
+	hook.Remove("PlayerInitialSpawn","__e2_MAP_HAS_NAV")
+end )
+
+local NULL_MESH = navmesh.GetNavAreaByID(0)
+local NULL_VEC = Vector()
+local NULL_TABLE = {}
+
+local aStarLoopCheck = 0
+local ASTAR_OP_MULTIPLIER = 10
+
+local Clamp = math.Clamp
+local floor = math.floor
+
+--------------------
+
+registerType("navmesh", "xnv", NULL_MESH,
+	nil,
+	nil,
+	function(retval)
+		if retval == nil then return end

```

Comment:
I'm not sure, but doesn't IsValid already cover this?
Also, does IsValid work on CNavMeshes?

Reply:
Yea the nil check is unnecessary. There's probably a lot of other places in e2 with this check but it can be removed. I think IsValid does work on them.